Sources/NautilusTelemetry/Tracing/URLTemplateMatcher.swift

```diff
@@ -0,0 +1,199 @@
+// Created by Jon Parise on 6/20/25.
+// Copyright © 2025 Airbnb Inc. All rights reserved.
+
+import Foundation
+
+// MARK: - URLTemplateMatcher
+
+/// Matches URLs against a collection of URL template patterns.
+///
+/// Template patterns describe a URL path and query string. They can use
+/// templated parameters like {} or {id} to match dynamic portions of the
+/// path and query parameter values (e.g. "/users/{id}?status={status}").
+///
+/// ## Template Syntax
+///
+/// - Path parameters: `/users/{userId}`
+/// - Query parameters: `?status={status}`
+/// - Literal query values: `?status=active`
+/// - Mixed: `/users/{id}?status={status}&verified=true`
+///
+/// ## Query Parameters
+///
+/// Query parameters can appear in any order to match. Any query parameters
+/// that aren't named in the template are ignored.
+///
+/// If a literal query parameter pair (e.g. "status=active") is included in
+/// the template, it must appear exactly that way in the URL for it to match.
+///
+/// ## Examples
+///
+/// ```swift
+/// let matcher = try URLTemplateMatcher(templates: [
```

Comment:
I think this is implied, but we should recommend the URLTemplateMatcher be initialized once, and not for each use.